runtime/compiler/compile/J9SymbolReferenceTable.cpp

```diff
@@ -1269,6 +1269,12 @@ static bool isSignatureTypeBool(const char *fieldSignature, int32_t len)
    return len == 1 && fieldSignature[0] == 'Z';
    }
 
+static bool isSignatureReturnTypeBool(const char *methodSignature, int32_t len)
+   {
+   // Method signature must end with ")Z" to have a boolean result type
+   return len > 1 && (')' == methodSignature[len-2]) && ('Z' == methodSignature[len-1]);
```

Comment:
We may want to have an assertion here to make sure `strlen(methodSignature)` and `len` is at least two.

Reply:
I think the value of `methodSignature` is not necessarily null-terminated, so I will just assert that `len > 1`

Reply:
Revised in commit https://github.com/eclipse/openj9/pull/4086/commits/9e34a9d6f838efda22f40c737d833e5c60f4ed00